ts_src/psetv2/signer.ts

```diff
@@ -0,0 +1,93 @@
+import { Transaction } from '../transaction';
+import { PartialSig } from './interfaces';
+import { Pset, ValidateSigFunction } from './pset';
+import { Updater } from './updater';
+import { isP2WPKH, isP2WSH } from './utils';
+
+export class Signer {
```

Comment:
I don't think it's necessary. The role of this signer is to add a partial signature to a pset input by prior validating it.
It may need to be changed for taproot inputs but the idea is to work on interface abstraction to support both cases.
We won't have a TaprootSigner, otherwise signing a partial transaction with taproot and non-taproot inputs would be a mess.